assets/js/src/welcome_wizard/steps_controller.jsx

```diff
@@ -8,57 +8,42 @@ import WelcomeWizardEmailCourseStep from './steps/email_course_step.jsx';
 import WelcomeWizardUsageTrackingStep from './steps/usage_tracking_step.jsx';
 import WelcomeWizardWooCommerceStep from './steps/woo_commerce_step.jsx';
 
-class WelcomeWizardStepsController extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      stepsCount: window.is_woocommerce_active ? 4 : 3,
-      shouldSetSender: !window.is_mp2_migration_complete,
-      loading: false,
-      sender: window.sender_data,
-      replyTo: window.reply_to_data,
-    };
-
-    this.finishWizard = this.finishWizard.bind(this);
-    this.updateSettings = this.updateSettings.bind(this);
-    this.activateTracking = this.activateTracking.bind(this);
-    this.updateSender = this.updateSender.bind(this);
-    this.updateReplyTo = this.updateReplyTo.bind(this);
-    this.submitSender = this.submitSender.bind(this);
-    this.showWooCommerceStepOrFinish = this.showWooCommerceStepOrFinish.bind(this);
-    this.componentDidUpdate();
-  }
-
-  componentDidUpdate() {
-    const step = parseInt(this.props.match.params.step, 10);
-    if (step > this.state.stepsCount || step < 1) {
-      this.props.history.push('/steps/1');
+const WelcomeWizardStepsController = (props) => {
+  const [stepsCount] = useState(window.is_woocommerce_active ? 4 : 3);
+  const [shouldSetSender] = useState(!window.is_mp2_migration_complete);
+  const [loading, setLoading] = useState(false);
+  const [sender, setSender] = useState(window.sender_data);
+  const [replyTo, setReplyTo] = useState(window.reply_to_data);
+
+  useEffect(() => {
```

Comment:
We need to execute this effect only when the `step` changes. So I would define the `step` before calling `useEffect` and pass it in the second argument like this:

```js
const step = parseInt(props.match.params.step, 10);
useEffect(() => {
  if (step > stepsCount || step < 1) {
    props.history.push('/steps/1');
  }
}, [step]);
```